src/nvim/menu.c

```diff
@@ -1301,11 +1301,15 @@ void ex_emenu(exarg_T *eap)
 
   /* Found the menu, so execute.
    * Use the Insert mode entry when returning to Insert mode. */
-  if (restart_edit
+  if (
+      ((State & INSERT) || restart_edit)
```

Comment:
1. AFAIK you need `get_real_state()` to check for visual mode, but not for insert mode
2. Just found it easier to read.